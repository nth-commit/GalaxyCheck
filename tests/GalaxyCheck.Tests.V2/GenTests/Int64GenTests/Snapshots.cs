using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int64GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IIntGen<long>>
            {
                { "Between_0_And_10", Gen.Int64().Between(0, 10) },
                { "Between_0_And_100", Gen.Int64().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int64().Between(0, 10000) },
                { "Between_Default", Gen.Int64() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IIntGen<long>>
            {
                { "Between_0_And_10", Gen.Int64().Between(0, 10) },
                { "Between_0_And_100", Gen.Int64().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
