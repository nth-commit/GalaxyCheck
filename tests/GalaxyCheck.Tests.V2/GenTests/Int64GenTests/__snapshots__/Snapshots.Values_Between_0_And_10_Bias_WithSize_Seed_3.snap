[
  0,
  1,
  1,
  0,
  1,
  0,
  1,
  0,
  0,
  1,
  1,
  1,
  0,
  1,
  0,
  1,
  0,
  1,
  1,
  1,
  0,
  3,
  4,
  5,
  5,
  3,
  3,
  2,
  2,
  3,
  2,
  0,
  5,
  0,
  2,
  1,
  1,
  3,
  1,
  5,
  1,
  8,
  10,
  3,
  8,
  4,
  2,
  4,
  0,
  3,
  2,
  0,
  8,
  3,
  2,
  6,
  5,
  7,
  0,
  0,
  6,
  10,
  0,
  2,
  7,
  10,
  7,
  8,
  2,
  8,
  10,
  2,
  5,
  9,
  10,
  5,
  1,
  10,
  9,
  2,
  7,
  4,
  10,
  9,
  10,
  10,
  10,
  6,
  10,
  10,
  10,
  10,
  4,
  10,
  3,
  9,
  10,
  10,
  5,
  10
]
